dali/imgcodec/decoders/nvjpeg/nvjpeg.cc

```diff
@@ -171,17 +173,52 @@ DecodeResult NvJpegDecoderInstance::DecodeImplTask(int thread_idx,
   } else {
     CUDA_CALL(nvjpegDecodeParamsSetROI(ctx.resources.params, 0, 0, -1, -1));
   }
+
+  auto orientation = opts.use_orientation ? JpegParser().GetInfo(in).orientation : Orientation{};
+  bool is_orientation_adjusted = orientation.rotate || orientation.flip_x || orientation.flip_y;
+
+  // We don't decode directly to YCbCr, since we want to control the YCbCr definition,
+  // which is different between general color conversion libraries (OpenCV) and
+  // what JPEG uses.
+  // JPEG files are always using bitdepth 8.
+  bool needs_processing = opts.format == DALI_YCbCr ||
+                          opts.dtype != DALI_UINT8 ||
+                          is_orientation_adjusted;
+
   try {
     ParseJpegSample(*in, opts, ctx);
+
     if (roi.use_roi()) {
       ctx.shape[0] = roi.shape()[0];
       ctx.shape[1] = roi.shape()[1];
```

Comment:
We need to use un-rotated size here - (as we do in ParseJpegSample). This is most certainly a bug and needs a follow-up work (this PR or otherwise).